libipl/libipl.C

```diff
@@ -359,3 +362,43 @@ void ipl_error_callback(const ipl_error_info& error)
 		return;
 	g_ipl_error_callback_fn(error);
 }
+
+void ipl_process_fapi_error(const fapi2::ReturnCode& fapirc,
+		struct pdbg_target *target, bool deconfig)
+{
+	if(fapirc == fapi2::FAPI2_RC_SUCCESS) {
+		ipl_error_callback(IPL_ERR_OK);
+	} 
+	else if(fapirc.getCreator() == fapi2::ReturnCode::CREATOR_PLAT) {
+		CDG_Target cdgTarget;
+		cdgTarget.callout_priority =
+		fapi2::plat_CalloutPriority_tostring(fapi2::CalloutPriorities::MEDIUM);
+
+		ATTR_PHYS_BIN_PATH_Type physBinPath;
+		uint32_t binPathElemCount = dtAttr::fapi2::ATTR_PHYS_BIN_PATH_ElementCount;
+		if (!pdbg_target_get_attribute(target, "ATTR_PHYS_BIN_PATH",
+			std::stoi(dtAttr::fapi2::ATTR_PHYS_BIN_PATH_Spec),
+				binPathElemCount, physBinPath)) {
+			ipl_log(IPL_ERROR, "Failed to read ATTR_PHYS_BIN_PATH for proc %s\n",
+				pdbg_target_path(target));
+			ipl_error_callback(IPL_ERR_ATTR_READ_FAIL);
+		}
+		else {
+			std::copy(physBinPath, physBinPath+binPathElemCount,
+			std::back_inserter(cdgTarget.target_entity_path));
+			cdgTarget.deconfigure = deconfig;
+
+			FFDC* ffdc = new FFDC();;
+			ffdc->hwp_errorinfo.cdg_targets.push_back(cdgTarget);
+			ffdc->ffdc_type = FFDC_TYPE_HWP;
+			uint32_t rc = fapirc;
```

Comment:
`fapi2::ReturnCode& fapirc` is the class in the ekb, right? is there any operator overloading?

Reply:
yes it is there